internal/jobservice/eventstojobs/eventstojobs.go

```diff
@@ -87,7 +87,11 @@ func (eventToJobService *EventsToJobService) streamCommon(ctx context.Context, t
 			select {
 			case <-ctx.Done():
 				log.Errorf("context is done on %s/%s", eventToJobService.queue, eventToJobService.jobSetId)
-				return nil
+				_, err := eventToJobService.jobServiceRepository.UnsubscribeJobSet(
```

Comment:
We need to actually clean up the the job set sub from the DB when we bail out. So do it.